src/services/hash.password.bcryptjs.ts

```diff
@@ -17,3 +20,28 @@ export async function hashPassword(
   const salt = await genSalt(rounds);
   return await hash(password, salt);
 }
+
+export interface PasswordHasher<T = string> {
+  hashPassword(password: T): Promise<T>;
+  comparePassword(providedPass: T, storedPass: T): Promise<boolean>;
+}
+
+export class BcryptHasher implements PasswordHasher<string> {
+  constructor(
+    @inject(PasswordHasherBindings.ROUNDS)
+    private readonly rounds: number,
+  ) {}
+
+  async hashPassword(password: string): Promise<string> {
+    const salt = await genSalt(this.rounds);
+    return await hash(password, salt);
+  }
+
+  async comparePassword(
+    providedPass: string,
+    storedPass: string,
+  ): Promise<boolean> {
+    const passwordIsMatched = await compare(providedPass, storedPass);
+    return passwordIsMatched;
```

Comment:
nitpick: can we do `return await compare(providedPass, storedPass)` here?

Reply:
@b-admike Sorry forgot to apply this change, I will remember to do it when refactor this example to use new authentication module.